laalaa/apps/advisers/tasks.py

```diff
@@ -54,7 +47,10 @@ def prime_geocoder_cache():
     Cache known postcode locations
     """
     for location in models.Location.objects.exclude(point__isnull=True):
-        geocode.cache[location.postcode] = location.point
+        cache.set(to_key(location.postcode.encode('utf-8')), {
+            'lon': location.point.x,
+            'lat': location.point.y
+        })
```

Comment:
If you cached `location.point` here instead of a dict, the rest of the code would be simpler